src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -1221,7 +1221,16 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
         case Some(jar: ZipArchive) => // We are in a jar
           val cl = new URLClassLoader(Array(jar.toURL), /*parent =*/ null)
           val path = file.path.stripSuffix(".class") + ".tasty"
-          val stream = cl.getResourceAsStream(path)
+
+          val tastyConnection = cl.getResource(path).openConnection()
```

Comment:
Looks like `getResource` can return null. I'm going to change up the code a bit to make it safer.